_posts/2023-07-17-elytron-oidc-client-scope.adoc

```diff
@@ -0,0 +1,153 @@
+
+Click on "Access Secured Servlet".
+
+Now, you’ll be redirected to Keycloak to log in. If you click on the url on the search bar, you will see the scope values specified in the `redirect-uri` field with the different scope values separated by a `+`. You will also notice that a new scope value, `openid`, has been added. This indicates that we are going to be using OpenID Connect to authenticate the user. 
+
+Log in with `Alice` and the password that you set when configuring Keycloak.
+
+Next, you’ll be redirected back to our application and you should see the "Secured Servlet" page. That means that we were able to successfully log in to our application using the Keycloak OpenID provider!
+
+This page will display the Current Principal, which is Alice and a list of claim values obtained using the scope values you configured. 
+
+Notice that there is no claims obtained using the `offline_access` scope. To learn more about what this scope value does, please refer to the https://openid.net/specs/openid-connect-core-1_0.html#OfflineAccess[ OpenID Documentation ]. 
+
+=== Invalid Scope Values
+
+Different OpenID providers have a set list of valid scope values and this list varies depending on the OpenID provider. Try changing the scope values to `INVALID_SCOPE`. 
+
+If you used the subsystem configuration, then navigate to the `WILDFLY_HOME/bin` folder and type in the following commands: 
+```
+./bin/jboss-cli.sh --connect --controller=127.0.0.1:10000
+
+/subsystem=elytron-oidc-client/secure-deployment=simple-webapp-oidc.war:write-attribute(name=scope, value=INVALID_SCOPE)
+```
+
+If you used the deployment configuration, then you can go back to the `oidc.json` file inside the `WEB-INF` folder to edit the scope values. 
+
+Deploy and access the webapp again. Since `INVALID_SCOPE` is not one of the acceptable scope values, you will now see a `Bad request` page instead of being redirected to the Keycloak login page. You will notice that the url now contains 
+`error=invalid_scope&error_description=Invalid+scopes`
+This indicates that your authentication request was rejected because it contains invalid scope values. 
+
+Once you are ready to restore your server back to what it was, please enter the following on your terminal: 
+```
+$PATH/TO/WILDFLY_HOME/bin/jboss-cli.sh --connect --controller=127.0.0.1:10000 --file=$PATH/TO/ELYTRON/EXAMPLES/elytron-oidc-client-scope/restore-elytron-configuration.cli
+```
+
+== Summary
+
+This example has demonstrated how to secure a web application deployed to WildFly using additional scope values on OpenID Connect. For more details on the `elytron-oidc-client` subsystem, please check out the https://docs.wildfly.org/26/Admin_Guide.html#Elytron_OIDC_Client[ documentation ] and for more details on OpenID Connect, checkout the https://openid.net/specs/openid-connect-core-1_0.html#ScopeClaims[ OpenID documentation ] and the documentation of your OpenID provider. 
```

Comment:
Just a note for later, we'll need to update the link to the WildFly documentation.

Reply:
Implemented changes till here. documentation link is changed to refer to the WildFly 29 docs, but the link will not work yet. 
Removed realm from deployment config, and had to add `public-client=true` otherwise it responded with forbidden error since we did not specify client secret.